drivers/SmartThings/philips-hue/src/handlers/lifecycle_handlers.lua

```diff
@@ -9,6 +9,31 @@ local StrayDeviceHelper = require "stray_device_helper"
 
 local utils = require "utils"
 
+local function check_parent_assigned_child_key(device)
+  local device_type = utils.determine_device_type(device)
+  local device_rid = utils.get_hue_rid(device)
+
+  if type(device_type) == "string" and type(device_rid) == "string" then
+    local expected_parent_assigned_child_key = string.format("%s:%s", device_type, device_rid)
+    if expected_parent_assigned_child_key ~= device.parent_assigned_child_key then
+      log.debug(
+        string.format(
+          "\n\nDevice [%s] had parent-assigned child key of %s but expected %s, requesting metadata update\n\n",
+          (device and device.label) or "unknown device",
+          device.parent_assigned_child_key,
+          expected_parent_assigned_child_key
+        )
+      )
+      -- updating parent_assigned_child_key in metadata isn't supported
+      -- on Lua Libs API versions before 11.
+      if require("version").api <= 10 then
+        return
+      end
+      device:try_update_metadata({ parent_assigned_child_key = expected_parent_assigned_child_key })
```

Comment:
So this function will essentially be a no-op until 0.54 FW? And we intend to add support for this in that timeframe?

Reply:
I can move the version number out of it doesn't land, but it's implemented on that PR I have open internally for Scripting Engine that you commented on where I need to update the firmware side structs.